pkg/controlplane/controller.go

```diff
@@ -172,16 +176,38 @@ func (c *Controller) Start() {
 	// than 1 minute for backward compatibility of failing the whole
 	// apiserver if we can't repair them.
 	wg := sync.WaitGroup{}
-	wg.Add(2)
+	wg.Add(1)
 
-	runRepairClusterIPs := func(stopCh chan struct{}) {
-		repairClusterIPs.RunUntil(wg.Done, stopCh)
-	}
 	runRepairNodePorts := func(stopCh chan struct{}) {
 		repairNodePorts.RunUntil(wg.Done, stopCh)
 	}
 
-	c.runner = async.NewRunner(c.RunKubernetesNamespaces, c.RunKubernetesService, runRepairClusterIPs, runRepairNodePorts)
+	wg.Add(1)
+	var runRepairClusterIPs func(stopCh chan struct{})
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MultiCIDRServiceAllocator) {
+		repairClusterIPs := servicecontroller.NewRepair(c.ServiceClusterIPInterval,
+			c.client.CoreV1(),
+			c.client.EventsV1(),
+			&c.ServiceClusterIPRange,
+			c.ServiceClusterIPRegistry,
+			&c.SecondaryServiceClusterIPRange,
+			c.SecondaryServiceClusterIPRegistry)
+		runRepairClusterIPs = func(stopCh chan struct{}) {
+			repairClusterIPs.RunUntil(wg.Done, stopCh)
+		}
+	} else {
+		repairClusterIPs := servicecontroller.NewRepairIPAddress(c.ServiceClusterIPInterval,
+			c.client,
+			&c.ServiceClusterIPRange,
+			&c.SecondaryServiceClusterIPRange,
+			c.informers.Core().V1().Services(),
+			c.informers.Networking().V1alpha1().IPAddresses(),
+		)
+		runRepairClusterIPs = func(stopCh chan struct{}) {
```

Comment:
This is the same in both legs of the if statement, why not put it outside the if statement?

Reply:
because this way I keep the Runner the same and reduce the diff, tomato tomatoe , no?
```
	c.runner = async.NewRunner(c.RunKubernetesNamespaces, c.RunKubernetesService, runRepairNodePorts, runRepairClusterIPs)
```

Reply:
See line 213, this doesn't move forward until the repair run once

Reply:
ack.